irods/session.py

```diff
@@ -354,11 +384,36 @@ def port(self):
         return self.pool.account.port
 
     @property
-    def server_version(self):
+    def server_version(self): return self._server_version()
+
+    def raw_server_version(self):
+        """Returns the same version tuple as iRODSSession's server_version property, but
+           does not require successful authentication.
+        """
+        import irods.connection
+        with self.clone().pool.no_auto_authenticate() as pool:
+            return irods.connection.Connection(
+                pool, pool.account
+            ).server_version
+
+    RAW_SERVER_VERSION = staticmethod(lambda s:s.raw_server_version())
+
+    def _server_version(self, version_func = None):
+        """The  server version can be retrieved by the usage
+            session._server_version()
+        with conditional substitution by another version by use of the environment variable:
+            PYTHON_IRODSCLIENT_REPORTED_SERVER_VERSION.
+
+        Also: if iRODSServer.RAW_SERVER_VERSION is passed in version_func, the true server
+        version can be accessed without first going through authentication.
+        Example:
+            ses = irods.helpers.make_session()
+            vsn = ses._server_version( ses.RAW_SERVER_VERSION )
```

Comment:
Does this cause the return type of `_server_version` to change? The word "raw" indicates to me that we are getting some more... "raw" than the default "server version". Is it just a different means of acquiring the same result (i.e. deriving the server version without authenticating)?

Reply:
no.... no actual change in the functions or return types, just some juggling going on there. Here, raw meant the connection gets cooked as little as possible before returning the version tuple.  I'm open to suggestions instead of raw.
But I have not come up with anything better myself. Possibilities are: 
   * immediate_server_version
   * server_version_without_auth
but I don't like either, and I don't like anything else I've thought of.

Reply:
@alanking - To explain why it's valuable: the new, irods version agnostic free function to write a pam .irodsA auth file (like iinit does) from the PRC needed to sense the server version , but the ` server_version ` property couldn't do so without first authenticating. That was an obvious no-go, so i made the "raw" version.  No authenticating , just get the server version number from the initial handshake.

Reply:
Right, understood. It's like the non-property way of acquiring the server version from the session, which is in some sense "raw". I don't dislike `server_version_without_auth`, personally. It's much clearer to me what the function is for. However, if the current name suits everybody else just fine, then it's fine.

Reply:
`server_version_without_auth` is good enough. I think the name is clear and reduces the possibility of confusion.

Let's roll with that.